app/assets/javascripts/reducers/users.js

```diff
@@ -23,12 +23,38 @@ const SORT_DESCENDING = {
 
 export default function users(state = initialState, action) {
   switch (action.type) {
-    case RECEIVE_USERS: return {
+    case RECEIVE_USERS: {
+      // Get the sorting key from the Redux store
+      let sort_key = state.sort.key;
+
+       // Initialize a variable to hold the user list. If there are existing users in the state,
+      //  use that list; otherwise, use the user list from the action data.
+      let user_list = state.users.length ? state.users : action.data.course.users;
+
+       // Check if the user list is empty in the state and if any user in the action data has 'real_name'
+       if (!state.users.length && action.data.course.users.some(user => user.real_name)) {
```

Comment:
I think this transformation ought to be extracted to a separate pure function (which can live in model_utils, along with the sortByKey function).

In addition to making the reducer logic easier to understand, this would let us test the sorting behavior in a more isolated way, which I think is important for this kind of thing where we may discover sorting edge cases over time that we need to handle and test for regressions.

Reply:
Okay sure :)